clients/python-wrapper/lakefs/cmd.py

```diff
@@ -0,0 +1,221 @@
+"""
+This module provides functionality to find and run the lakeFS and lakectl binaries.
+
+The module provides the following functions:
+- find_or_download_binary: Find the lakeFS binary in 
+    $PATH or ~/.lakefs/bin, or download it if not found.
+- run_binary: Run the lakeFS binary with the provided arguments.
+"""
+
+import os
+import subprocess
+import sys
+import shutil
+import platform
+import tarfile
+from typing import Optional
+import io
+import urllib.request
+import json
+from collections import namedtuple
+import zipfile
+
+try:
+    import tqdm
+except ImportError:
+    tqdm = None
+
+
+BINARY_DOWNLOAD_DIR = '~/.lakefs/bin'
```

Comment:
This might not be writeable.  And I know that /tmp/ might be a different filesystem, which introduces another point of failure.  Finally, using a predictable download name as the target for an executable is a known vulnerability.
I _think_ Python [tempfile](https://docs.python.org/3/library/tempfile.html) can solve all of these issues.

Reply:
This doesn't download into a predictable location - the release is downloaded and extracted into memory and the binary is placed in a known location. I don't see how this differs from what package managers (e.g. homebrew) do.

I think it's safe to assume that for the vast majority of users, being able to create a directory in their home folder and writing a file to it is possible.

Reply:
Not sure it is possible, but I wish it was installed as part of the python bin environemnt. so when I'm inside venv the binary will be placed inside <venv>/bin of my env.